addons/mail/models/mail_thread.py

```diff
@@ -1091,8 +1202,14 @@ def message_process(self, model, message, custom_values=None,
                          msg_dict.get('email_from'), msg_dict.get('to'), msg_dict.get('message_id'))
             return False
 
+        if self._routing_detect_loop_from_headers(message, msg_dict):
```

Comment:
If this does what I think, we're going to blindly ignore quite a lot of valid emails. We don't have any reason to believe that they are related to a loop at this stage.

There are legit cases where users *want* to receive those kind of emails, e.g. because they've subscribed to an automated service that sends emails to a dedicated alias, and they're even paying for that. Or because they're a salesperson who care about the auto-reply that says their prospect is on maternity leave for 3 months.

People will do all sorts of crazy setups with emails, with automated forwards, mailing-lists and whatnot. As long as there's no loop, they should be free to do it, without having emails mysteriously disappearing ;-)